### ISSUE TITLE
How to filter updates with allowed_updates?

### ISSUE THREAD

**Issue by sidelux** [yagop/node-telegram-bot-api#243]
I read days ago new api version where telegram now take possibility to filter incoming update with allowed_updates in getUpdates method.
How can i write this method correctly to receive only text messages and edited messages?

Thanks a lot!

**Comment by GochoMugo** [comment]
This feature would certainly be useful :+1:

---

First, the current implementation of the [TelegramBot#getUpdates()](https://github.com/yagop/node-telegram-bot-api#telegrambotgetupdatestimeout-limit-offset--promise) does **not** allow us to pass the `allowed_updates` parameter. This is being addressed in PR #245.

---

Second, we need a way to provide parameters to be re-used during polling, in our [Constructor](https://github.com/yagop/node-telegram-bot-api#new-telegrambottoken-options). For example,

```js
const bot = new TelegramBot(token, {
  polling: {
    // parameters reused in each request during polling
    getUpdates: {
      "allowed_updates": ["message", "edited_channel_post", "callback_query"],
    },
  },
});
```

(Will work on this second part, and send PR asap)

**Comment by GingerPlusPlus** [comment]
@GochoMugo or, instead of allowing users specify which updates they wanna receive, what about building that list based on the updates they listen to?

```
this.allowedUpdates = [];
on(event, callback) {
    super(event, callback);
    this.addAllowedUpdate(event);
}
addAllowedUpdate(event) {
    if (!this.allowedUpdates.includes(event) {
        this.allowedUpdates.push(event);
    }
}
onText() {
    // other stuff
  addAllowedUpdate(event);
}
```

**Comment by GochoMugo** [comment]
I have just added this enhancement (in commit [e75d51ca8fe46059e78a74e1ece7787c881157cb](https://github.com/yagop/node-telegram-bot-api/commit/e75d51ca8fe46059e78a74e1ece7787c881157cb)). Please try out the updated master branch's code to see this feature in action.

@GingerPlusPlus Sorry, I felt your way being too complex. We would have to coordinate when polling starts and when are done scanning for the allowed updates. However, we can discuss more on it, in depth (preferably in an entirely new issue).

**Comment by GochoMugo** [comment]
Shipped in **v0.27.0**.

Thanks @sidelux and @GingerPlusPlus for contributing to this feature.
